WordImpress/Give
# Issues/1849

## Description

This PR will resolve #1849 
This PR can affect all possible place where we are sanitizing and showing amount like email, donation form, donations listing, donor list etc.

## How Has This Been Tested?



Note: Before following the testing instruction you have to run updates otherwise you will get the same issue. Please also test this without updates, for checking are you getting any issue with currency formatting. I added support for INR currency in amount formatting and human readable amount function.

**Settings**
During testing trying different combinations of decimal ( `,`, `.` ) and thousand ( `,`, `.`, ` `) separators with decimal precision (0-5 [ 6 precision is core standard now]).
**Note:** Core will validate decimal and thousand separators from now. here are few cases:
1. Decimal and thousands separator can not be same.
2. Number decimal can not be non-zero if decimal separator is set to empty.
3. Six number of the decimal is core standard.

To test this PR, you can test following thing:
1. Add a new donation from front end and check following things
   a. Email receipt
   b. Confirmation page receipt
   c. Donation in list and detail page
   d. Log entry
   e. Donor count
   f. Forn earnings
   g. Reports  
2. Perform following the action on donation and review step1 sublist from point `c` again.
   a. Update donation amount
   b. Change donor
   c. Chnage form 
3. Review step 1 and 2 with following addon also:
   a. Payment gateways: stripe. paypal etc.
   b. Recurring donation
   c. Manual donation
   b. Email reports
   c. Fee recovery


## Types of changes

Bug fix (non-breaking change which fixes an issue)



## Checklist:
